src/Service/Drush.php

```diff
@@ -97,19 +97,20 @@ public function getLegacyAliasFiles()
      *
      * @return string|false
      *   The Drush version, or false if it cannot be determined.
-     *
-     * @throws DependencyMissingException
-     *   If Drush is not installed.
      */
     public function getVersion($reset = false)
     {
         static $version;
         if (!$reset && isset($version)) {
             return $version;
         }
-        $this->ensureInstalled();
 
-        $version = $this->shellHelper->execute([$this->getDrushExecutable(), 'version', '--format=string'], $this->getHomeDir());
+        $version = $this->shellHelper->execute(
+            [$this->getDrushExecutable(), 'version', '--format=string'],
+            // Run the version check in the home directory, to avoid a Drupal
+            // installation giving a misleading result.
+            $this->getHomeDir()
```

Comment:
Hm. This _is_ running the site-local Drush (if available), because the getDrushExecutable() returns an absolute path, but it's running it with the home directory as the CWD. Although I can't remember exactly what problem this solves anymore (it seems like a workaround for earlier Drush 9 versions again).